packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -356,15 +389,33 @@ export const GridRootStyles = styled('div', {
         },
       },
     },
+    [`& .${gridClasses['container--top']}, & .${gridClasses['container--bottom']}`]: {
+      '[role=row]': {
+        background: 'var(--unstable_DataGrid-containerBackground)',
```

Comment:
Rows in the column header should have `minWidth: 100%`:

<img width="870" src="https://github.com/mui/mui-x/assets/13808724/dc1dabee-05b8-4287-bead-6c7576b48a46">

Reply:
Where is this example?

Reply:
This one: https://deploy-preview-10059--material-ui-x.netlify.app/x/react-data-grid/filtering/

But it's reproducible in any demo that has vertical scroll and the column total width smaller than the grid width